From d7755a1773677fdc00c45a892ed4e182180b929e Mon Sep 17 00:00:00 2001
From: "Maneendra,Pokuri Naga" <pnmaneendra@maxlinear.com>
Date: Thu, 27 Feb 2025 12:56:02 +0530
Subject: [PATCH] WLANRTSYS-84595 Ignoring NULL ifaces in rnr IE creation for
 MLO

FIXUP to WLANRTSYS-62845 MLO changes in iwlwav-hostap-ng:
AP MLD Auth Assoc and eapol changes.
---
 src/ap/ieee802_11.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 9ca16aa07..504869a29 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -8323,6 +8323,11 @@ static size_t hostapd_eid_rnr_multi_iface_len(struct hostapd_data *hapd,
 		iface = hapd->iface->interfaces->iface[i];
 		bool ap_mld = false;
 
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		if (!iface)
+			continue;
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
+
 #ifdef CONFIG_IEEE80211BE
 		if (hapd->conf->mld_ap && iface->bss[0]->conf->mld_ap &&
 		    hapd->conf->mld_id == iface->bss[0]->conf->mld_id)
@@ -8591,6 +8596,11 @@ static u8 * hostapd_eid_rnr_multi_iface(struct hostapd_data *hapd, u8 *eid,
 		iface = hapd->iface->interfaces->iface[i];
 		bool ap_mld = false;
 
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+		if (!iface)
+			continue;
+#endif /* CONFIG_IEEE80211BE_MXL_MLO */
+
 #ifdef CONFIG_IEEE80211BE
 		if (hapd->conf->mld_ap && iface->bss[0]->conf->mld_ap &&
 		    hapd->conf->mld_id == iface->bss[0]->conf->mld_id)
-- 
2.43.0

